tests/test_scalars.py

```diff
@@ -708,6 +805,293 @@ def test_timestamp() -> None:
     check(assert_type(ts.month, int), int)
     check(assert_type(ts.nanosecond, int), int)
     check(assert_type(ts.second, int), int)
-    check(assert_type(ts.tzinfo, None), type(None))
+    check(assert_type(ts.tzinfo, Optional[dt.tzinfo]), type(None))
     check(assert_type(ts.value, int), int)
     check(assert_type(ts.year, int), int)
+
+    ts = pd.Timestamp("2000-1-1")
+    check(assert_type(ts + pd.Timedelta(days=1), pd.Timestamp), pd.Timestamp)
+    check(assert_type(ts + dt.timedelta(days=1), pd.Timestamp), pd.Timestamp)
+    check(assert_type(pd.Timedelta(days=1) + ts, pd.Timestamp), pd.Timestamp)
+    check(assert_type(dt.timedelta(days=1) + ts, pd.Timestamp), pd.Timestamp)
+    check(assert_type(ts + 3 * Day(), pd.Timestamp), pd.Timestamp)
+    check(assert_type(3 * Day() + ts, pd.Timestamp), pd.Timestamp)
+    check(
+        assert_type(ts + pd.TimedeltaIndex([1, 2, 3], "D"), pd.DatetimeIndex),
+        pd.DatetimeIndex,
+    )
+    check(
+        assert_type(ts + pd.Series([1, 2], dtype="timedelta64[ns]"), "TimestampSeries"),
+        pd.Series,
+    )
+    np_td64_arr: npt.NDArray[np.timedelta64] = np.array([1, 2], dtype="timedelta64[ns]")
+    np_dt64_arr: npt.NDArray[np.datetime64] = np.array(
+        [1, 2, 3], dtype="datetime64[ns]"
+    )
+    check(
+        assert_type(ts + np_td64_arr, npt.NDArray[np.datetime64]),
+        np.ndarray,
+    )
+    check(
+        assert_type(pd.TimedeltaIndex([1, 2, 3], "D") + ts, pd.DatetimeIndex),
+        pd.DatetimeIndex,
+    )
+    check(
+        assert_type(pd.Series([1, 2], dtype="timedelta64[ns]") + ts, "TimestampSeries"),
+        pd.Series,
+    )
+
+    check(assert_type(ts - pd.Timedelta(days=1), pd.Timestamp), pd.Timestamp)
+    check(assert_type(ts - dt.timedelta(days=1), pd.Timestamp), pd.Timestamp)
+    check(assert_type(ts - 3 * Day(), pd.Timestamp), pd.Timestamp)
+    check(
+        assert_type(ts - pd.TimedeltaIndex([1, 2, 3], "D"), pd.DatetimeIndex),
+        pd.DatetimeIndex,
+    )
```

Comment:
should be checking `object - ts` as well for each of the tests that check `object - ts`

Reply:
Subtraction is not symmetric here